sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -30,9 +30,21 @@ template <class To, class From> To cast(From value) {
   return (To)(value);
 }
 
+// Names of USM functions that are queried from OpenCL
+const char clHostMemAllocName[] = "clHostMemAllocINTEL";
+const char clDeviceMemAllocName[] = "clDeviceMemAllocINTEL";
+const char clSharedMemAllocName[] = "clSharedMemAllocINTEL";
+const char clMemFreeName[] = "clMemFreeINTEL";
+const char clSetKernelArgMemPointerName[] = "clSetKernelArgMemPointerINTEL";
+const char clEnqueueMemsetName[] = "clEnqueueMemsetINTEL";
+const char clEnqueueMemcpyName[] = "clEnqueueMemcpyINTEL";
+const char clEnqueueMigrateMemName[] = "clEnqueueMigrateMemINTEL";
+const char clEnqueueMemAdviseName[] = "clEnqueueMemAdviseINTEL";
+const char clGetMemAllocInfoName[] = "clGetMemAllocInfoINTEL";
+
 // USM helper function to get an extension function pointer
-template <typename T>
-pi_result getExtFuncFromContext(pi_context context, const char *func, T *fptr) {
+template <const char *FuncName, typename T>
```

Comment:
I though string literals cannot be template arguments, not sure how this is different and why it works.
@garimagu is adding similar functionality for all of PI interfaces, but there she is introducing enum argument rather than "const char*". Not sure which one is better.

Reply:
I think because it's not actually the literal value being used, just the pointer to the global which is basically a weird int? @rolandschulz would know better.

Reply:
Didn't know about the feature of the language, very interesting.